src/kernel/syscalls/sys_read.c

```diff
@@ -37,32 +35,34 @@ ssize_t sys_read_keyboard(void *buffer, size_t nbytes) {
     // I need to get access to the current task
     task_t *current_task = current_executing_thread->parent_task;
     VmmInfo vmm_info = current_task->vmm_data;
+    pending_operation_t *new_pending_operation = kmalloc(sizeof(pending_operation_t));
+    userspace_buffer_t *userspace_buffer = kmalloc(sizeof(userspace_buffer_t));
```

Comment:
I cant remember if `kmalloc()` can fail and return NULL, if it can it's worth checking for that here and returning an error to userspace if it does.